pkg/daemon/daemon.go

```diff
@@ -400,16 +402,31 @@ func (dn *Daemon) nodeUpdateHandler(old, new interface{}) {
 		return
 	}
 	dn.node = node.DeepCopy()
+
 	nodes, err := dn.nodeLister.List(labels.Everything())
 	if err != nil {
+		glog.Errorf("nodeUpdateHandler(): failed to list nodes: %v", err)
 		return
 	}
-	for _, node := range nodes {
-		if node.GetName() != dn.name && (node.Annotations[annoKey] == annoDraining || node.Annotations[annoKey] == annoMcpPaused) {
+
+	// Checking if other nodes are draining
+	for _, otherNode := range nodes {
+		if otherNode.GetName() == dn.name {
+			continue
+		}
+
+		drainingAnnotationValue := otherNode.Annotations[annoKey]
+		if drainingAnnotationValue == annoDraining || drainingAnnotationValue == annoMcpPaused {
+			glog.V(2).Infof("nodeUpdateHandler(): node is not drainable as [%s] has [%s == %s] ", otherNode.Name, annoKey, drainingAnnotationValue)
 			dn.drainable = false
 			return
 		}
 	}
+
+	if !dn.drainable {
+		glog.V(2).Infof("nodeUpdateHandler(): node is drainable")
```

Comment:
nit: node is now drainable ? 

as you only log the transition from false to true here

Reply:
agree